### PR TITLE
Delete entry reader temporary local files before exit

### LINKED ISSUE #5287: [Bug]: Scala EntryRecordReader doesn't close sstable iterator

### PR DESCRIPTION
**Pull request by eden-ohana**
UGC fails on error `java.io.IOException: Too many open files` when creating a new EntryRecordReader.

deleteOnExit() deletes the file only on exit of the machine.

fixes #5287 #5288

### CODE REVIEW

**Comment by arielshaqed** [comment]
I don't understand how _deleting_ a file reduces the number of open file descriptors.  So, this change may be good -- but I am afraid that I do not understand how it can fix the issue described in the PR.

On POSIX, if you delete an open file, it remains open and you can read it and write to it.  Deleting does not free a file descriptor.

File: clients/spark/core/src/main/scala/io/treeverse/clients/LakeFSInputFormat.scala [hunk 1]
```diff
@@ -123,7 +124,9 @@ class EntryRecordReader[Proto <: GeneratedMessage with scalapb.Message[Proto]](
 
   override def getCurrentValue = new WithIdentifier(item.id, item.message, rangeID)
 
-  override def close() = Unit
+  override def close(): Unit = {
+    sstableReader.close()
```
Comment (arielshaqed) [inline clients/spark/core/src/main/scala/io/treeverse/clients/LakeFSInputFormat.scala:128]: I think we _still_ **also** want to delete the file: deleteOnExit says to delete the file when the JVM terminates. For a Spark worker, that can be a loooooong while.

**Review by arielshaqed — CHANGES_REQUESTED** [review]
Thanks, this is a very good catch of a file descriptor leak!  But note that your earlier explicit deletion _also_ makes a lot of sense: Otherwise the disk on the worker fills up _until the JVM exits_.  Spark workers run for a long while, and there is not point in filling up their disks.

So I think 🤡  I want both 🥇 🥈

File: clients/spark/core/src/main/scala/io/treeverse/clients/LakeFSInputFormat.scala (hunk 1 above)
Comment (eden-ohana) [reply clients/spark/core/src/main/scala/io/treeverse/clients/LakeFSInputFormat.scala:129]: added both

**Review by arielshaqed — APPROVED** [review]
Thanks!  Very nice job, should run forever without loading the disk on the cluster _or_ the open file descriptors.

**Pull request merged**